support/cas-server-support-oauth/src/main/java/org/apereo/cas/config/CasOAuth20Configuration.java

```diff
@@ -242,6 +244,17 @@ public Config oauthSecConfig() {
         val clientList = oauthSecConfigClients();
         val config = new Config(OAuth20Utils.casOAuthCallbackUrl(casProperties.getServer().getPrefix()), clientList);
         config.setSessionStore(oauthDistributedSessionStore());
+        val csrfMatcher = new CsrfTokenGeneratorMatcher(new DefaultCsrfTokenGenerator());
```

Comment:
Is there a scenario where one would want to update/adjust the CSRF settings? Would it make sense to skip the cookie stuff and just disable or enable the CSRF matcher?

Reply:
The place where I was seeing the cookie blocked was when developers were hitting an app they were developing that was authenticating via CAS OIDC to a backend on a different domain (not localhost). In that case (in the development environment) I was setting the samesite policy to None which I think is the old default. The developers have gotten around it by using Firefox which lets you change the default (from Lax to None) and they just don't use Chrome when working on that app. So in terms of wanting to adjust the settings, I can see wanting to adjust the SameSite policy and I always prefer a path on cookies rather than having them at the top level domain.